packages/server/core/src/base/adapters/node/bff.ts

```diff
@@ -0,0 +1,46 @@
+import { isWebOnly } from '@modern-js/utils';
+import {
+  BindRenderHandleOptions,
+  getRenderHandler,
+} from '../../../base/renderHandler';
+import { ServerBaseOptions } from '../../../base';
+import { createMiddlewareCollecter, getRuntimeEnv } from '../../libs/utils';
+import { ServerBase } from '../../serverBase';
+import { ServerNodeMiddleware } from './hono';
+
+export const bindBFFHandler = async (
+  server: ServerBase,
+  options: ServerBaseOptions & BindRenderHandleOptions,
+) => {
+  const prefix = options.config.bff.prefix || '/api';
+  const { httpMethodDecider } = options.config.bff;
+  const runtimeEnv = getRuntimeEnv();
+  if (runtimeEnv !== 'node') {
+    return;
+  }
+  const { getMiddlewares, ...collector } = createMiddlewareCollecter();
+  const { runner } = server;
+  await runner.gather(collector);
+
+  const webOnly = await isWebOnly();
+  let handler: ServerNodeMiddleware;
+  const renderHandler = getRenderHandler(options);
+  if (webOnly) {
+    handler = async (c, next) => {
+      c.body('');
```

Comment:
Should we give a 404 response if `webOnly`?

Reply:
No, 404 will cause an error, which is not what users expect.

Reply:
Got it